src/lightning/data/utilities/shuffle.py

```diff
@@ -12,8 +12,9 @@ def _intra_node_chunk_shuffle(
     current_epoch: int,
 ) -> List[int]:
     chunk_indexes_per_nodes: Any = [[] for _ in range(distributed_env.num_nodes)]
+    process_per_node = distributed_env.world_size // distributed_env.num_nodes
```

Comment:
maybe add this attr to `_DistributedEnv ` ?

Reply:
Yes, this is a good idea! Feel free to do so.